lib/datadog/tracing.rb

```diff
@@ -0,0 +1,198 @@
+      #
+      # See {Datadog::Configuration::Settings} for all available options, defaults, and
+      # available environment variables for configuration.
+      #
+      # @yieldparam [Datadog::Configuration::Settings] c the mutable configuration object
+      # @return [void]
+      # @public_api
+      def configure(&block)
+        Datadog.configure(&block)
+      end
+
+      # (see Datadog::Tracer#active_trace)
+      # @public_api
+      def active_trace
+        tracer.active_trace
+      end
+
+      # (see Datadog::Tracer#active_span)
+      # @public_api
+      def active_span
+        tracer.active_span
+      end
+
+      # (see Datadog::TraceSegment#keep!)
+      # If no trace is active, no action is taken.
+      # @public_api
+      def keep!
+        trace = active_trace
+        active_trace.keep! if trace
+      end
+
+      # (see Datadog::TraceSegment#reject!)
+      # If no trace is active, no action is taken.
+      # @public_api
+      def reject!
+        trace = active_trace
+        active_trace.reject! if trace
+      end
+
+      # (see Datadog::Tracer#active_correlation)
+      # @public_api
+      def correlation
+        tracer.active_correlation
+      end
+
+      # Textual representation of {.correlation}, which can be
+      # added to individual log lines in order to correlate them with the active
+      # trace.
+      #
+      # Example:
+      #
+      # ```
+      # MyLogger.log("#{Datadog::Tracing.log_correlation}] My log message")
+      # # dd.env=prod dd.service=billing dd.version=13.8 dd.trace_id=545847825299552251 dd.span_id=711755234730770098 My log message
+      # ```
+      #
+      # @return [String] correlation information
+      # @public_api
+      def log_correlation
+        correlation.to_log_format
+      end
+
+      # Gracefully shuts down the tracer.
+      #
+      # The public tracing API will still respond to method calls as usual
+      # but might not internally perform the expected internal work after shutdown.
+      #
+      # This avoids errors being raised across the host application
+      # during shutdown while allowing for the graceful decommission of resources.
+      #
+      # {.shutdown!} cannot be reversed.
+      # @public_api
+      def shutdown!
+        components.shutdown!
+      end
+
+      # The global integration registry.
+      #
+      # This registry holds a reference to all integrations available
+      # to the tracer.
+      #
+      # Integrations registered in the {.registry} can be activated as follows:
+      #
+      # ```
+      # Datadog.configure do |c|
+      #   c.use :my_registered_integration, **my_options
+      # end
+      # ```
+      #
+      # New integrations can be registered by implementing the {Datadog::Contrib::Integration} interface.
+      #
+      # @return [Datadog::Contrib::Registry]
+      # @!attribute [r] registry
+      # @public_api
+      def registry
+        Datadog::Contrib::REGISTRY
+      end
```

Comment:
Hmm... This one is somewhat weird as well; if integrations are expected to only be something that applies to tracing, should we fix their namespace to indicate that?

Reply:
Are you talking about this line specifically `Datadog::Contrib::REGISTRY`, or something else?

Reply:
Yeap, I meant the registry. If we're going to rename it, yeah, no issues on my side, thanks for pointing that out :)